shared/AppInsightsCore/src/JavaScriptSDK.Interfaces/IFeatureOptIn.ts

```diff
@@ -0,0 +1,21 @@
+import { FeatureOptInMode } from "../JavaScriptSDK.Enums/FeatureOptInEnums";
+
+export interface IFeatureOptInDetails {
+    /**
+     * sets feature opt-in mode
+     * @default undefined
+     */
+    mode?: FeatureOptInMode;
+    /**
+     * set values of current feature
+     * @default undefined
+     */
+    cfgValue?: any;
```

Comment:
Based on how we are using this now we need to define this as a map now.

`{ [key: string]: any }`